core/src/connection.rs

```diff
@@ -30,16 +33,14 @@ impl ConnectionId {
     /// This is primarily useful for creating connection IDs
     /// in test environments. There is in general no guarantee
     /// that all connection IDs are based on non-negative integers.
+    #[deprecated(since = "0.39.0", note = "Use `ConnectionId::next` instead.")]
     pub fn new(id: usize) -> Self {
```

Comment:
Are we aware of any users of `ConnectionId::new` outside of this repository?

Providing both `new` and `next` removes the guarantee of `next`, namely to create unique `ConnectionId`s.

In case there are no users, how about removing `new` in favor of `next` directly?

Reply:
I was going to deprecate it first because it doesn't matter until we merge https://github.com/libp2p/rust-libp2p/pull/3328.

Reply:
Let's merge https://github.com/libp2p/rust-libp2p/pull/3221 first anyway.